src/Runner.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -119,6 +119,17 @@ public async Task ConfigureAsync(CommandSettings command)
                     // Determine the service deployment type based on connection data. (Hosted/OnPremises)
                     runnerSettings.IsHostedServer = runnerSettings.GitHubUrl == null || IsHostedServer(new UriBuilder(runnerSettings.GitHubUrl));
 
+                    // Warn if the Actions server url and GHES server url has different Host
```

Comment:
nice

Reply:
Chatted offline...

Example `actionsServerUrl` is `https://my-ghes/_services/pipelines/[...]`
Example `githubServerUrl` is `https://my-ghes`

Consider adding ^ example into the comments in the source code

Should we just fail? Should use hosts file, or DNS, or proxy bypass to fix routing. Server doesn't support multiple host names.